### ISSUE TITLE
Nothing happens after script completes

### ISSUE THREAD

**Issue by camsterguy** [LukeZGD/iOS-OTA-Downgrader#58]
my iPhone 5,1 when trying to downgrade just stays in DFU mode after the script says "restore completed"

Rebooting it out of DFU still boots into ios 10. Does the script not actually perform the restore process? Just confused. Thanks!

**Comment by camsterguy** [comment]
[Log] Extracting IPSW... 
[Log] Proceeding to idevicerestore... 
./restore.sh: line 373:  5287 Killed: 9               $idevicerestore -y -e -w $IPSW.ipsw

[Log] Restoring done! 
[Log] Downgrade script done!

**Comment by LukeZGD** [comment]
cd to the directory of the script and try running: `sudo codesign --sign - --force --deep resources/tools/idevicerestore`

**Comment by LukeZGD** [comment]
fixed w/ latest commit

**Comment by UInt2048** [comment]
@LukeZGD Experienced this in latest commit. Do I need to apply that command, and if so what do I do next - rerun the script? I'm also on iPhone5,1 running on macOS 11.0 beta (20A5395g).

The phone is stuck in DFU and won't seem to leave or display anything.

<details>
<summary>Log</summary>

```
Matthew-MacBookPro16-2:IPSW matthewbenedict$ ./restore.sh

******* iOS-OTA-Downgrader ******* 
   Downgrader script by LukeZGD    

[Log] Finding device in normal mode... 
* Platform: macos 11.0 
* HardwareModel: n41ap 
* ProductType: iPhone5,1 
* ProductVersion: 10.3.4 
* UniqueChipID (ECID): 2244574471118 

*** Main Menu *** 
[Input] Select an option: 
1) Downgrade device		 4) (Re-)Install Dependencies
2) Save OTA blobs		 5) (Any other key to exit)
3) Just put device in kDFU mode
#? 1
[Input] Select iOS version: 
1) iOS 8.4.1		    3) (Any other key to exit)
2) Other
#? 1
[Log] Option: Downgrade 
[Input] Jailbreak the selected iOS version? (y/N):  n
By default, iOS-OTA-Downgrader now flashes the iOS 8.4.1 baseband to iPhone5,1 
Flashing the latest baseband is still available as an option but beware of problems it may cause 
There are potential network issues that with the latest baseband when used on iOS 8.4.1 
[Input] Flash the latest baseband? (y/N) (press ENTER when unsure):  
[Log] Saving 8.4.1 blobs with tsschecker... 
Version: b9d193aa6e6d24421094873c830692d02d8b32f5 - 304
libfragmentzip version: 0.59-542a470d7be248681dba71d0f04e7dc8c2718b73
[TSSC] manually specified ECID to use, parsed "2244574471118" to dec:2244574471118 hex:20a9b1063ce
[TSSC] opening resources/manifests/BuildManifest_iPhone5,1_8.4.1.plist
[WARNING] [TSSC] could not get id0 for installType=Erase. Using fallback installType=Update since user did not specify installType manually
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... success
[Error] [TSSR] Error: could not get id0 for installType=Erase
Saved shsh blobs!

iOS 8.4.1 for device iPhone5,1 IS being signed!
[Log] Successfully saved 8.4.1 blobs. 
[Log] Verifying IPSW... 
[Log] Extracting iBSS from IPSW... 
Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
  inflating: saved/iPhone5,1/iBSS.n41.RELEASE.dfu  
[Log] Patching iBSS... 
bind(): Address already in use
Error creating socket: Address already in use
[Log] Copying stuff to device via SSH... 
* Make sure OpenSSH/Dropbear is installed on the device! 
* Enter root password of your iOS device when prompted, default is 'alpine' 
accepted connection, fd = 4
waiting for connection
Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
The authenticity of host '[127.0.0.1]:2222 ([127.0.0.1]:2222)' can't be established.
RSA key fingerprint is SHA256:cJpaFZ2hNIb8k+YhlvpY7BN24pZkkf2wI/vKAovAM8Y.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
root@127.0.0.1's password: 
kloader_hgsp                                                      100%   52KB   3.4MB/s   00:00    
pwnediBSS                                                         100%   76KB   4.8MB/s   00:00    
pwn.sh                                                            100%   53    14.8KB/s   00:00    
recv failed: Connection reset by peer
[Log] Entering kDFU mode... 

* Press POWER or HOME button when screen goes black on the device 
[Log] Finding device in DFU mode... 
recv failed: Connection reset by peer
accepted connection, fd = 4
waiting for connection
Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
root@127.0.0.1's password: 
recv failed: Connection reset by peer
recv failed: Operation not permitted
Connection to 127.0.0.1 closed by remote host.
[Log] Found device in DFU mode. 
./restore.sh: line 320: kill: (3051) - No such process
[Log] Extracting IPSW... 
[Log] Preparing for futurerestore... (Enter root password of your PC/Mac when prompted) 
Password:
Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
  inflating: saved/iPhone5,1/Mav5-8.02.00.Release.bbfw  
[Log] Proceeding to futurerestore... 
Version: b99eb8140d8e6c23f34e950102bb79e61c72384d - 152
Libipatcher Version: f32e41d850f51448bd6c588ead9c7d6455733f3c - 44
Odysseus Support: yes
[INFO] 32bit device detected
futurerestore init done
reading ticket 2244574471118_iPhone5,1_8.4.1-12H321_c4267934602fbd1bb04a2c201d9908cfaf573b1a.shsh2 done
[Error] could not open file BuildManifest.plist
[Error] failed to load BasebandManifest[Error] Fail code=-14
Failed with errorcode=-14

[Log] Restoring done! 
[Log] Stopping local server... (Enter root password of your PC/Mac when prompted) 
Serving HTTP on 0.0.0.0 port 80 ...
[Log] Downgrade script done! 
```
</details>

**Comment by LukeZGD** [comment]
> @LukeZGD Experienced something very similar in latest commit. Do I need to apply that command, and if so what do I do next - rerun the script? I'm also on iPhone5,1 running on macOS 11.0 beta (20A5395g).
> 
> The phone is stuck in DFU and won't seem to leave or display anything.
> 
> Thank you so much for this tool, and I can't wait to use it to get a working downgrade!
> Log
> 
> ```
> Matthew-MacBookPro16-2:IPSW matthewbenedict$ ./restore.sh
> 
> ******* iOS-OTA-Downgrader ******* 
>    Downgrader script by LukeZGD    
> 
> [Log] Finding device in normal mode... 
> * Platform: macos 11.0 
> * HardwareModel: n41ap 
> * ProductType: iPhone5,1 
> * ProductVersion: 10.3.4 
> * UniqueChipID (ECID): 2244574471118 
> 
> *** Main Menu *** 
> [Input] Select an option: 
> 1) Downgrade device		 4) (Re-)Install Dependencies
> 2) Save OTA blobs		 5) (Any other key to exit)
> 3) Just put device in kDFU mode
> #? 1
> [Input] Select iOS version: 
> 1) iOS 8.4.1		    3) (Any other key to exit)
> 2) Other
> #? 1
> [Log] Option: Downgrade 
> [Input] Jailbreak the selected iOS version? (y/N):  n
> By default, iOS-OTA-Downgrader now flashes the iOS 8.4.1 baseband to iPhone5,1 
> Flashing the latest baseband is still available as an option but beware of problems it may cause 
> There are potential network issues that with the latest baseband when used on iOS 8.4.1 
> [Input] Flash the latest baseband? (y/N) (press ENTER when unsure):  
> [Log] Saving 8.4.1 blobs with tsschecker... 
> Version: b9d193aa6e6d24421094873c830692d02d8b32f5 - 304
> libfragmentzip version: 0.59-542a470d7be248681dba71d0f04e7dc8c2718b73
> [TSSC] manually specified ECID to use, parsed "2244574471118" to dec:2244574471118 hex:20a9b1063ce
> [TSSC] opening resources/manifests/BuildManifest_iPhone5,1_8.4.1.plist
> [WARNING] [TSSC] could not get id0 for installType=Erase. Using fallback installType=Update since user did not specify installType manually
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... success
> [Error] [TSSR] Error: could not get id0 for installType=Erase
> Saved shsh blobs!
> 
> iOS 8.4.1 for device iPhone5,1 IS being signed!
> [Log] Successfully saved 8.4.1 blobs. 
> [Log] Verifying IPSW... 
> [Log] Extracting iBSS from IPSW... 
> Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
>   inflating: saved/iPhone5,1/iBSS.n41.RELEASE.dfu  
> [Log] Patching iBSS... 
> bind(): Address already in use
> Error creating socket: Address already in use
> [Log] Copying stuff to device via SSH... 
> * Make sure OpenSSH/Dropbear is installed on the device! 
> * Enter root password of your iOS device when prompted, default is 'alpine' 
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> The authenticity of host '[127.0.0.1]:2222 ([127.0.0.1]:2222)' can't be established.
> RSA key fingerprint is SHA256:cJpaFZ2hNIb8k+YhlvpY7BN24pZkkf2wI/vKAovAM8Y.
> Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
> Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
> root@127.0.0.1's password: 
> kloader_hgsp                                                      100%   52KB   3.4MB/s   00:00    
> pwnediBSS                                                         100%   76KB   4.8MB/s   00:00    
> pwn.sh                                                            100%   53    14.8KB/s   00:00    
> recv failed: Connection reset by peer
> [Log] Entering kDFU mode... 
> 
> * Press POWER or HOME button when screen goes black on the device 
> [Log] Finding device in DFU mode... 
> recv failed: Connection reset by peer
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> root@127.0.0.1's password: 
> recv failed: Connection reset by peer
> recv failed: Operation not permitted
> Connection to 127.0.0.1 closed by remote host.
> [Log] Found device in DFU mode. 
> ./restore.sh: line 320: kill: (3051) - No such process
> [Log] Extracting IPSW... 
> [Log] Preparing for futurerestore... (Enter root password of your PC/Mac when prompted) 
> Password:
> Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
>   inflating: saved/iPhone5,1/Mav5-8.02.00.Release.bbfw  
> [Log] Proceeding to futurerestore... 
> Version: b99eb8140d8e6c23f34e950102bb79e61c72384d - 152
> Libipatcher Version: f32e41d850f51448bd6c588ead9c7d6455733f3c - 44
> Odysseus Support: yes
> [INFO] 32bit device detected
> futurerestore init done
> reading ticket 2244574471118_iPhone5,1_8.4.1-12H321_c4267934602fbd1bb04a2c201d9908cfaf573b1a.shsh2 done
> [Error] could not open file BuildManifest.plist
> [Error] failed to load BasebandManifest[Error] Fail code=-14
> Failed with errorcode=-14
> 
> [Log] Restoring done! 
> [Log] Stopping local server... (Enter root password of your PC/Mac when prompted) 
> Serving HTTP on 0.0.0.0 port 80 ...
> [Log] Downgrade script done! 
> ```

Try again w/ latest commit 7349895

**Comment by LukeZGD** [comment]
> @LukeZGD Experienced something very similar in latest commit. Do I need to apply that command, and if so what do I do next - rerun the script? I'm also on iPhone5,1 running on macOS 11.0 beta (20A5395g).
> 
> The phone is stuck in DFU and won't seem to leave or display anything.
> 
> Thank you so much for this tool, and I can't wait to use it to get a working downgrade!
> Log
> 
> ```
> Matthew-MacBookPro16-2:IPSW matthewbenedict$ ./restore.sh
> 
> ******* iOS-OTA-Downgrader ******* 
>    Downgrader script by LukeZGD    
> 
> [Log] Finding device in normal mode... 
> * Platform: macos 11.0 
> * HardwareModel: n41ap 
> * ProductType: iPhone5,1 
> * ProductVersion: 10.3.4 
> * UniqueChipID (ECID): 2244574471118 
> 
> *** Main Menu *** 
> [Input] Select an option: 
> 1) Downgrade device		 4) (Re-)Install Dependencies
> 2) Save OTA blobs		 5) (Any other key to exit)
> 3) Just put device in kDFU mode
> #? 1
> [Input] Select iOS version: 
> 1) iOS 8.4.1		    3) (Any other key to exit)
> 2) Other
> #? 1
> [Log] Option: Downgrade 
> [Input] Jailbreak the selected iOS version? (y/N):  n
> By default, iOS-OTA-Downgrader now flashes the iOS 8.4.1 baseband to iPhone5,1 
> Flashing the latest baseband is still available as an option but beware of problems it may cause 
> There are potential network issues that with the latest baseband when used on iOS 8.4.1 
> [Input] Flash the latest baseband? (y/N) (press ENTER when unsure):  
> [Log] Saving 8.4.1 blobs with tsschecker... 
> Version: b9d193aa6e6d24421094873c830692d02d8b32f5 - 304
> libfragmentzip version: 0.59-542a470d7be248681dba71d0f04e7dc8c2718b73
> [TSSC] manually specified ECID to use, parsed "2244574471118" to dec:2244574471118 hex:20a9b1063ce
> [TSSC] opening resources/manifests/BuildManifest_iPhone5,1_8.4.1.plist
> [WARNING] [TSSC] could not get id0 for installType=Erase. Using fallback installType=Update since user did not specify installType manually
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... success
> [Error] [TSSR] Error: could not get id0 for installType=Erase
> Saved shsh blobs!
> 
> iOS 8.4.1 for device iPhone5,1 IS being signed!
> [Log] Successfully saved 8.4.1 blobs. 
> [Log] Verifying IPSW... 
> [Log] Extracting iBSS from IPSW... 
> Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
>   inflating: saved/iPhone5,1/iBSS.n41.RELEASE.dfu  
> [Log] Patching iBSS... 
> bind(): Address already in use
> Error creating socket: Address already in use
> [Log] Copying stuff to device via SSH... 
> * Make sure OpenSSH/Dropbear is installed on the device! 
> * Enter root password of your iOS device when prompted, default is 'alpine' 
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> The authenticity of host '[127.0.0.1]:2222 ([127.0.0.1]:2222)' can't be established.
> RSA key fingerprint is SHA256:cJpaFZ2hNIb8k+YhlvpY7BN24pZkkf2wI/vKAovAM8Y.
> Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
> Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
> root@127.0.0.1's password: 
> kloader_hgsp                                                      100%   52KB   3.4MB/s   00:00    
> pwnediBSS                                                         100%   76KB   4.8MB/s   00:00    
> pwn.sh                                                            100%   53    14.8KB/s   00:00    
> recv failed: Connection reset by peer
> [Log] Entering kDFU mode... 
> 
> * Press POWER or HOME button when screen goes black on the device 
> [Log] Finding device in DFU mode... 
> recv failed: Connection reset by peer
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 47 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> root@127.0.0.1's password: 
> recv failed: Connection reset by peer
> recv failed: Operation not permitted
> Connection to 127.0.0.1 closed by remote host.
> [Log] Found device in DFU mode. 
> ./restore.sh: line 320: kill: (3051) - No such process
> [Log] Extracting IPSW... 
> [Log] Preparing for futurerestore... (Enter root password of your PC/Mac when prompted) 
> Password:
> Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
>   inflating: saved/iPhone5,1/Mav5-8.02.00.Release.bbfw  
> [Log] Proceeding to futurerestore... 
> Version: b99eb8140d8e6c23f34e950102bb79e61c72384d - 152
> Libipatcher Version: f32e41d850f51448bd6c588ead9c7d6455733f3c - 44
> Odysseus Support: yes
> [INFO] 32bit device detected
> futurerestore init done
> reading ticket 2244574471118_iPhone5,1_8.4.1-12H321_c4267934602fbd1bb04a2c201d9908cfaf573b1a.shsh2 done
> [Error] could not open file BuildManifest.plist
> [Error] failed to load BasebandManifest[Error] Fail code=-14
> Failed with errorcode=-14
> 
> [Log] Restoring done! 
> [Log] Stopping local server... (Enter root password of your PC/Mac when prompted) 
> Serving HTTP on 0.0.0.0 port 80 ...
> [Log] Downgrade script done! 
> ```

@UInt2048 Try again with latest commit 7349895

**Comment by UInt2048** [comment]
Just ran on latest commit 7fadc9f9c721a788639cc09186756071cce0b1d2 - this time using GitHub Desktop to clone to ensure I don't have to do anything with Gatekeeper so it's not that. It (not-so) silently failed again with a different code.

<details>
<summary>Log</summary>

```
******* iOS-OTA-Downgrader ******* 
   Downgrader script by LukeZGD    

[Log] Downloading firmware... 
Clonando en 'firmware'...
remote: Enumerating objects: 689, done.
remote: Counting objects: 100% (689/689), done.
remote: Compressing objects: 100% (360/360), done.
remote: Total 689 (delta 138), reused 681 (delta 131), pack-reused 0
Recibiendo objetos: 100% (689/689), 164.07 KiB | 2.65 MiB/s, listo.
Resolviendo deltas: 100% (138/138), listo.
[Log] Downloading ipwndfu... 
Clonando en 'ipwndfu'...
remote: Enumerating objects: 459, done.
remote: Total 459 (delta 0), reused 0 (delta 0), pack-reused 459
Recibiendo objetos: 100% (459/459), 1.88 MiB | 7.76 MiB/s, listo.
Resolviendo deltas: 100% (235/235), listo.
[Log] Finding device in normal mode... 
* Platform: macos 11.0 
* HardwareModel: n41ap 
* ProductType: iPhone5,1 
* ProductVersion: 10.3.4 
* UniqueChipID (ECID): 2244574471118 

*** Main Menu *** 
[Input] Select an option: 
1) Downgrade device		 4) (Re-)Install Dependencies
2) Save OTA blobs		 5) (Any other key to exit)
3) Just put device in kDFU mode
#? 1
[Input] Select iOS version: 
1) iOS 8.4.1		    3) (Any other key to exit)
2) Other
#? 1
[Log] Option: Downgrade 
[Input] Jailbreak the selected iOS version? (y/N):  n
By default, iOS-OTA-Downgrader now flashes the iOS 8.4.1 baseband to iPhone5,1 
Flashing the latest baseband is still available as an option but beware of problems it may cause 
There are potential network issues that with the latest baseband when used on iOS 8.4.1 
[Input] Flash the latest baseband? (y/N) (press ENTER when unsure):  
[Log] Saving 8.4.1 blobs with tsschecker... 
Version: b9d193aa6e6d24421094873c830692d02d8b32f5 - 304
libfragmentzip version: 0.59-542a470d7be248681dba71d0f04e7dc8c2718b73
[TSSC] manually specified ECID to use, parsed "2244574471118" to dec:2244574471118 hex:20a9b1063ce
[TSSC] opening resources/manifests/BuildManifest_iPhone5,1_8.4.1.plist
[WARNING] [TSSC] could not get id0 for installType=Erase. Using fallback installType=Update since user did not specify installType manually
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... success
[Error] [TSSR] Error: could not get id0 for installType=Erase
Saved shsh blobs!

iOS 8.4.1 for device iPhone5,1 IS being signed!
[Log] Successfully saved 8.4.1 blobs. 
[Log] iOS 8.4.1 IPSW cannot be found. 
* If you already downloaded the IPSW, did you put it in the same directory as the script? 
* Do NOT rename the IPSW as the script will fail to detect it 
[Log] Downloading IPSW... (Press Ctrl+C to cancel) 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1580M  100 1580M    0     0  49.4M      0  0:00:31  0:00:31 --:--:-- 48.0M
[Log] Verifying IPSW... 
[Log] Extracting iBSS from IPSW... 
Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
  inflating: saved/iPhone5,1/iBSS.n41.RELEASE.dfu  
[Log] Patching iBSS... 
[Log] Copying stuff to device via SSH... 
* Make sure OpenSSH/Dropbear is installed on the device! 
* Enter root password of your iOS device when prompted, default is 'alpine' 
waiting for connection
accepted connection, fd = 4
waiting for connection
Requesting connecion to USB device handle 19 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
root@127.0.0.1's password: 
kloader_hgsp                                                      100%   52KB   3.9MB/s   00:00    
pwnediBSS                                                         100%   76KB   6.2MB/s   00:00    
pwn.sh                                                            100%   53    24.8KB/s   00:00    
recv failed: Connection reset by peer
[Log] Entering kDFU mode... 

* Press POWER or HOME button when screen goes black on the device 
[Log] Finding device in DFU mode... 
recv failed: Connection reset by peer
accepted connection, fd = 4
waiting for connection
Requesting connecion to USB device handle 19 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
root@127.0.0.1's password: 
recv failed: Connection reset by peer
recv failed: Operation not permitted
Connection to 127.0.0.1 closed by remote host.
[Log] Found device in DFU mode. 
[Log] Extracting IPSW... 
[Log] Preparing for futurerestore... (Enter root password of your PC/Mac when prompted) 
Password:
[Log] Downloading baseband... 
Initializing download for Firmware/Mav5-8.02.00.Release.bbfw...
Serving HTTP on 0.0.0.0 port 80 ...
Downloading: 100%
Initializing download for BuildManifest.plist...
Downloading: 100%
[Log] Proceeding to futurerestore... 
Version: b99eb8140d8e6c23f34e950102bb79e61c72384d - 152
Libipatcher Version: f32e41d850f51448bd6c588ead9c7d6455733f3c - 44
Odysseus Support: yes
[INFO] 32bit device detected
futurerestore init done
reading ticket 2244574471118_iPhone5,1_8.4.1-12H321_98eef3b2d5d68566a4936b04e1ae73727296a230.shsh2 done
Did set sep+baseband path and firmware
[WARNING] failed to read BasebandGoldCertID from device! Is it already in recovery?
[WARNING] using tsschecker's fallback to get BasebandGoldCertID. This might result in invalid baseband signing status information
[TSSC] opening BuildManifest.plist
WARNING: Unable to find BbSkeyId node
[TSSR] User specified to request only a Baseband ticket.
ERROR: Unable to get BasebandFirmware node
ERROR: Unable to find required BbGoldCertId in parameters
Request URL set to https://gs.apple.com/TSS/controller?action=2
TSS server returned: STATUS=94&MESSAGE=This device isn't eligible for the requested build.
ERROR: TSS request failed (status=94, message=This device isn't eligible for the requested build.)
Sending TSS request attempt 1... [Error] baseband firmware isn't signed
Failed with errorcode=-3

[Log] Restoring done! 
[Log] Stopping local server... (Enter root password of your PC/Mac when prompted) 
[Log] Downgrade script done! 
```
</details>

**Comment by LukeZGD** [comment]
> @LukeZGD Just ran on latest commit [7fadc9f](https://github.com/LukeZGD/iOS-OTA-Downgrader/commit/7fadc9f9c721a788639cc09186756071cce0b1d2) - this time using GitHub Desktop to clone to ensure I don't have to do anything with Gatekeeper so it's not that. It (not-so) silently failed again with a different code.
> Log
> 
> ```
> ******* iOS-OTA-Downgrader ******* 
>    Downgrader script by LukeZGD    
> 
> [Log] Downloading firmware... 
> Clonando en 'firmware'...
> remote: Enumerating objects: 689, done.
> remote: Counting objects: 100% (689/689), done.
> remote: Compressing objects: 100% (360/360), done.
> remote: Total 689 (delta 138), reused 681 (delta 131), pack-reused 0
> Recibiendo objetos: 100% (689/689), 164.07 KiB | 2.65 MiB/s, listo.
> Resolviendo deltas: 100% (138/138), listo.
> [Log] Downloading ipwndfu... 
> Clonando en 'ipwndfu'...
> remote: Enumerating objects: 459, done.
> remote: Total 459 (delta 0), reused 0 (delta 0), pack-reused 459
> Recibiendo objetos: 100% (459/459), 1.88 MiB | 7.76 MiB/s, listo.
> Resolviendo deltas: 100% (235/235), listo.
> [Log] Finding device in normal mode... 
> * Platform: macos 11.0 
> * HardwareModel: n41ap 
> * ProductType: iPhone5,1 
> * ProductVersion: 10.3.4 
> * UniqueChipID (ECID): 2244574471118 
> 
> *** Main Menu *** 
> [Input] Select an option: 
> 1) Downgrade device		 4) (Re-)Install Dependencies
> 2) Save OTA blobs		 5) (Any other key to exit)
> 3) Just put device in kDFU mode
> #? 1
> [Input] Select iOS version: 
> 1) iOS 8.4.1		    3) (Any other key to exit)
> 2) Other
> #? 1
> [Log] Option: Downgrade 
> [Input] Jailbreak the selected iOS version? (y/N):  n
> By default, iOS-OTA-Downgrader now flashes the iOS 8.4.1 baseband to iPhone5,1 
> Flashing the latest baseband is still available as an option but beware of problems it may cause 
> There are potential network issues that with the latest baseband when used on iOS 8.4.1 
> [Input] Flash the latest baseband? (y/N) (press ENTER when unsure):  
> [Log] Saving 8.4.1 blobs with tsschecker... 
> Version: b9d193aa6e6d24421094873c830692d02d8b32f5 - 304
> libfragmentzip version: 0.59-542a470d7be248681dba71d0f04e7dc8c2718b73
> [TSSC] manually specified ECID to use, parsed "2244574471118" to dec:2244574471118 hex:20a9b1063ce
> [TSSC] opening resources/manifests/BuildManifest_iPhone5,1_8.4.1.plist
> [WARNING] [TSSC] could not get id0 for installType=Erase. Using fallback installType=Update since user did not specify installType manually
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... success
> [Error] [TSSR] Error: could not get id0 for installType=Erase
> Saved shsh blobs!
> 
> iOS 8.4.1 for device iPhone5,1 IS being signed!
> [Log] Successfully saved 8.4.1 blobs. 
> [Log] iOS 8.4.1 IPSW cannot be found. 
> * If you already downloaded the IPSW, did you put it in the same directory as the script? 
> * Do NOT rename the IPSW as the script will fail to detect it 
> [Log] Downloading IPSW... (Press Ctrl+C to cancel) 
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100 1580M  100 1580M    0     0  49.4M      0  0:00:31  0:00:31 --:--:-- 48.0M
> [Log] Verifying IPSW... 
> [Log] Extracting iBSS from IPSW... 
> Archive:  iPhone5,1_8.4.1_12H321_Restore.ipsw
>   inflating: saved/iPhone5,1/iBSS.n41.RELEASE.dfu  
> [Log] Patching iBSS... 
> [Log] Copying stuff to device via SSH... 
> * Make sure OpenSSH/Dropbear is installed on the device! 
> * Enter root password of your iOS device when prompted, default is 'alpine' 
> waiting for connection
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 19 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> root@127.0.0.1's password: 
> kloader_hgsp                                                      100%   52KB   3.9MB/s   00:00    
> pwnediBSS                                                         100%   76KB   6.2MB/s   00:00    
> pwn.sh                                                            100%   53    24.8KB/s   00:00    
> recv failed: Connection reset by peer
> [Log] Entering kDFU mode... 
> 
> * Press POWER or HOME button when screen goes black on the device 
> [Log] Finding device in DFU mode... 
> recv failed: Connection reset by peer
> accepted connection, fd = 4
> waiting for connection
> Requesting connecion to USB device handle 19 (serial: 3aa617c1b02ec58343a730ca05528321eca20828), port 22
> run_ctos_loop: fd = 4
> run_stoc_loop: fd = 4
> root@127.0.0.1's password: 
> recv failed: Connection reset by peer
> recv failed: Operation not permitted
> Connection to 127.0.0.1 closed by remote host.
> [Log] Found device in DFU mode. 
> [Log] Extracting IPSW... 
> [Log] Preparing for futurerestore... (Enter root password of your PC/Mac when prompted) 
> Password:
> [Log] Downloading baseband... 
> Initializing download for Firmware/Mav5-8.02.00.Release.bbfw...
> Serving HTTP on 0.0.0.0 port 80 ...
> Downloading: 100%
> Initializing download for BuildManifest.plist...
> Downloading: 100%
> [Log] Proceeding to futurerestore... 
> Version: b99eb8140d8e6c23f34e950102bb79e61c72384d - 152
> Libipatcher Version: f32e41d850f51448bd6c588ead9c7d6455733f3c - 44
> Odysseus Support: yes
> [INFO] 32bit device detected
> futurerestore init done
> reading ticket 2244574471118_iPhone5,1_8.4.1-12H321_98eef3b2d5d68566a4936b04e1ae73727296a230.shsh2 done
> Did set sep+baseband path and firmware
> [WARNING] failed to read BasebandGoldCertID from device! Is it already in recovery?
> [WARNING] using tsschecker's fallback to get BasebandGoldCertID. This might result in invalid baseband signing status information
> [TSSC] opening BuildManifest.plist
> WARNING: Unable to find BbSkeyId node
> [TSSR] User specified to request only a Baseband ticket.
> ERROR: Unable to get BasebandFirmware node
> ERROR: Unable to find required BbGoldCertId in parameters
> Request URL set to https://gs.apple.com/TSS/controller?action=2
> TSS server returned: STATUS=94&MESSAGE=This device isn't eligible for the requested build.
> ERROR: TSS request failed (status=94, message=This device isn't eligible for the requested build.)
> Sending TSS request attempt 1... [Error] baseband firmware isn't signed
> Failed with errorcode=-3
> 
> [Log] Restoring done! 
> [Log] Stopping local server... (Enter root password of your PC/Mac when prompted) 
> [Log] Downgrade script done! 
> ```

@UInt2048 Yeah I forgot something there, sorry about that, you may try again with latest commit e97b846

**Comment by UInt2048** [comment]
@LukeZGD I was able to restore on that commit successfully. However, I finally decided that I wanted to flash latest baseband, so that might affect the outcome.

**Comment by LukeZGD** [comment]
> @LukeZGD I was able to restore on that commit successfully. However, I finally decided that I wanted to flash latest baseband, so that might affect the outcome.

The fixes I applied are for 8.4.1 baseband, so both options should work properly niw
